### DISCUSSION TITLE
catch response body with ClientResponseError

### DISCUSSION THREAD

**Discussion by a5r0n** [aio-libs/aiohttp#5566] [Q&A] [answered]
HI all
 it's possible to catch the response body even if ClientResponseError raised?
 lot of API's respond with error message  along with status code,
 and json (or text)  response helps with error message..

idea with pseudo code:

```python
async with session.get(url) as resp:
    try:
        resp.raise_for_status()
        return await resp.json()
    except ClientResponseError as e:
        logger.debug(await resp.json())
```

**Comment by Dreamsorcerer** [comment, marked as answer]
Looks like the response is released when raised, so not something that is currently supported.

Probably easiest to try without `raise_for_status()`.
